tests/manager/sokovan/scheduler/test_scheduler.py

```diff
@@ -0,0 +1,572 @@
+"""Tests for the Scheduler class allocation methods."""
+
+import uuid
+from decimal import Decimal
+from typing import Optional
+from unittest.mock import AsyncMock, MagicMock
+
+import pytest
+
+from ai.backend.common.types import (
+    AccessKey,
+    AgentId,
+    ClusterMode,
+    ResourceSlot,
+    SessionId,
+    SessionTypes,
+)
+from ai.backend.manager.sokovan.scheduler.scheduler import (
+    Scheduler,
+    SchedulerArgs,
+    SchedulerRepository,
+    SchedulingConfig,
+)
+from ai.backend.manager.sokovan.scheduler.selectors.exceptions import AgentSelectionError
+from ai.backend.manager.sokovan.scheduler.selectors.selector import (
+    AgentInfo,
+    AgentSelectionConfig,
+)
+from ai.backend.manager.sokovan.scheduler.types import (
+    KernelWorkload,
+    SessionWorkload,
+)
+
+
+def create_session_workload(
+    session_id: Optional[SessionId] = None,
+    cluster_mode: ClusterMode = ClusterMode.SINGLE_NODE,
+    kernels: Optional[list[KernelWorkload]] = None,
+    designated_agent: Optional[AgentId] = None,
+    **kwargs,
```

Comment:
how about specifying signatures rather than using kwargs?

Reply:
I will leave it as is since it is test code.